hyperstyle/src/python/review/inspectors/issue_configs.py

```diff
@@ -0,0 +1,221 @@
+import logging
+from dataclasses import dataclass
+from typing import Callable, Dict, Optional, Pattern, Tuple, Union
+
+from hyperstyle.src.python.review.inspectors.common import is_fstring
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class IssueDescriptionParser:
+    """
+    Parser for an issue description.
+
+    The ``converter`` can be:
+
+    - A function --
+      In this case, all groups obtained from the description using the ``regexp`` will be converted using this function.
+
+    - A dictionary --
+      In this case, the corresponding converter will be applied to the named group obtained by the ``regexp``.
+      All groups in the ``regexp`` must be named, otherwise unnamed groups will be skipped.
+      For those groups for which no converter is defined, the default converter will be used.
+
+    By default, the converter is equal to the ``str`` function.
+    """
+
+    regexp: Pattern[str]
+    converter: Union[Callable, Dict[str, Callable]] = str
+
+    def parse(self, description: str) -> Optional[Tuple]:
+        """
+        Parse the description into a tuple of converted regex groups.
+
+        :return: A tuple of converted regex groups. If there is an error during parsing, None will be returned.
+        """
+
+        match = self.regexp.search(description)
+        if match is None or not match.groups():
+            logger.error(f'Unable to parse ("{self.regexp.pattern}") the description: {description}')
+            return None
+
+        try:
+            if isinstance(self.converter, dict):
+                args = tuple(self.converter.get(name, str)(group) for name, group in match.groupdict().items())
+            else:
+                args = tuple(self.converter(group) for group in match.groups())
+        except Exception as exception:
+            logger.error(f'Unable to convert the groups: {exception}')
+            return None
+
+        return args
+
+
+@dataclass(frozen=True)
+class IssueConfig:
+    """
+    Custom config for a code issue.
+
+    Required fields:
+
+    - ``origin_class`` -- An origin class of issue.
+    - ``new_description`` -- A new description for the issue. Can be a format string, but in this case the ``parser``
+      must be specified.
+
+    Optional fields:
+
+    - ``parser`` -- A description parser.
+
+    If you just want to replace the issue description with a static one, specify only the ``new_description``.
+    Otherwise, the ``new_description`` must be a format string, the ``parser`` must be specified and
+    the number of groups in the regex must match the number of fields in the ``new_description``.
+    """
+
+    origin_class: str
+
+    new_description: Optional[str] = None
+    parser: Optional[IssueDescriptionParser] = None
+
+    def __post_init__(self):
+        if self.new_description is None:
```

Comment:
This check looks strange since new_description: Optional[str] = None can be None....